### DISCUSSION TITLE
Automated submissions

### DISCUSSION THREAD

**Discussion by sbfnk** [hubverse-org/.github#2] [Ideas]
For the European COVID-19 hub we implemented an [automated submissions framework](https://github.com/european-modelling-hubs/covid19-forecast-hub-europe-submissions). It would be great to make this part of the hubverse, either as a set of instructions for teams on how to implement this, or as a set of tools for hub maintainers to facilitate it.

Ideally we might want to guide participants to implement their own autosubmissions so as not to be running stale/abandoned models, but we were unable to find a solution that would allow people to open a PR from a fork through GitHub Actions. The somewhat unsatisfactory solution we landed on instead was to have a central submission repo (the one you link) that would either run code for participants (if simple) or else pick up the file from a DropBox/web/other link and then submit it. Because this lived in the same organisation as the hub repo it could open PRs from a local branch.

@bisaloo correct me if this is wrong, please. Also tagging @seabbs for info/further insight as you have been relying heavily on automatic submissions.

**Comment by Bisaloo** [comment]
Yes, this was our assessment at the time. Note that I no longer believe it's accurate. It **should** be possible to non-interactively create a PR from a fork with [`gh pr create`](https://cli.github.com/manual/gh_pr_create).
